hardwarelibrary/communication/sutter.py

```diff
@@ -0,0 +1,149 @@
+        if it is not alread initialized. On failure, it will warn and shutdown."""
+        try:
+            if self.port is None:
+                self.initializeDevice()
+            
+            self.port.write(commandBytes)
+
+        except Exception as err:
+            print('Error when sending command: {0}'.format(err))
+            self.shutdownDevice()
+
+    def readReply(self, size, format) -> tuple:
+        """ The function to read a reply from the endpoint. It will initialize the device 
+        if it is not already initialized. On failure, it will warn and shutdown. 
+        It will unpack the reply into a tuple.
+        """
+        try:
+            if self.port is None:
+                self.initializeDevice()
+
+            replyBytes = self.port.read(size)
+            if len(replyBytes) == size:
+                theTuple = unpack(format, replyBytes)
+                return theTuple
+
+        except Exception as err:
+            print('Error when reading reply: {0}'.format(err))
+            self.shutdownDevice()
+            return None
+
+    def positionInMicrosteps(self) -> (int,int,int):
+        """ Returns the position in microsteps """
+        commandBytes = pack('<cc', b'C', b'\r')
+        self.sendCommand(commandBytes)
+        return self.readReply(size=14, format='<clllc')[1:-1]
+
+    def moveInMicrostepsTo(self, position):
+        """ Move to a position in microsteps """
+        x,y,z  = position
+        commandBytes = pack('<clllc', b'M', int(x), int(y), int(z), b'\r')
+        self.sendCommand(commandBytes)
+        #self.readReply(size=1, format='<c')
+    
+    def position(self) -> (float, float, float):
+        """ Returns the position in microns """
+
+        position = self.positionInMicrosteps()
+        if position is not None:
+            return (position[0]/self.microstepsPerMicrons, 
+                    position[1]/self.microstepsPerMicrons,
+                    position[2]/self.microstepsPerMicrons)
+        else:
+            return None
+
+    def moveTo(self, position):
+        """ Move to a position in microns """
+        #if not self.port.is_open:
+            #self.port.open()
+        x,y,z  = position
+        positionInMicrosteps = (x*self.microstepsPerMicrons, 
+                                y*self.microstepsPerMicrons,
+                                z*self.microstepsPerMicrons)
+        self.moveInMicrostepsTo(positionInMicrosteps)
+        #self.port.close()
+
+    def moveBy(self, delta) -> bool:
+        #Move by a delta displacement (dx, dy, dz) from current position in microns
+        #if not self.port.is_open:
+            #self.port.open()
+        dx,dy,dz  = delta
+        position = self.position()
+        if position is not None:
+            x,y,z = position
+            self.moveTo((x+dx, y+dy, z+dz))
+        #self.port.close()
+
+
+if __name__ == "__main__":
```

Comment:
This should be elsewhere.  Where is the test file?

Reply:
I can create a new file with the loop...